kernel: bump 5.15 to 5.15.104

graysky2 opened the pull request:
All patches automatically rebased.

Build system: x86_64
Build-tested: bcm2711/RPi4B, ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: bcm2711/RPi4B, ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod

graysky2 commented:
As-is, building fails for:
```
CONFIG_TARGET_ramips=y
CONFIG_TARGET_ramips_mt7621=y
CONFIG_TARGET_ramips_mt7621_DEVICE_tplink_archer-a6-v3=y
```

Here is the failure:
```
...
make[6]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[7]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[8]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[9]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
  CC      drivers/net/ethernet/mediatek/mtk_eth_soc.o
drivers/net/ethernet/mediatek/mtk_eth_soc.c: In function 'mtk_mac_config':
drivers/net/ethernet/mediatek/mtk_eth_soc.c:503:9: error: 'mcr_cur' undeclared (first use in this function)
  503 |         mcr_cur = mtk_r32(mac->hw, MTK_MAC_MCR(mac->id));
      |         ^~~~~~~
drivers/net/ethernet/mediatek/mtk_eth_soc.c:503:9: note: each undeclared identifier is reported only once for each function it appears in
drivers/net/ethernet/mediatek/mtk_eth_soc.c:504:9: error: 'mcr_new' undeclared (first use in this function)
  504 |         mcr_new = mcr_cur;
      |         ^~~~~~~
drivers/net/ethernet/mediatek/mtk_eth_soc.c: In function 'mtk_mac_finish':
drivers/net/ethernet/mediatek/mtk_eth_soc.c:531:22: warning: unused variable 'mcr_new' [-Wunused-variable]
  531 |         u32 mcr_cur, mcr_new;
      |                      ^~~~~~~
drivers/net/ethernet/mediatek/mtk_eth_soc.c:531:13: warning: unused variable 'mcr_cur' [-Wunused-variable]
  531 |         u32 mcr_cur, mcr_new;
      |             ^~~~~~~
make[9]: *** [scripts/Makefile.build:289: drivers/net/ethernet/mediatek/mtk_eth_soc.o] Error 1
make[9]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[8]: *** [scripts/Makefile.build:552: drivers/net/ethernet/mediatek] Error 2
make[8]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[7]: *** [scripts/Makefile.build:552: drivers/net/ethernet] Error 2
make[7]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[6]: *** [scripts/Makefile.build:552: drivers/net] Error 2
make[6]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[5]: *** [Makefile:1912: drivers] Error 2
make[5]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[4]: *** [Makefile:23: /scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103/.modules] Error 2
make[4]: Leaving directory '/scratch/union/target/linux/ramips'
make[3]: *** [Makefile:11: compile] Error 2
make[3]: Leaving directory '/scratch/union/target/linux'
time: target/linux/compile#2.52#0.71#3.13
    ERROR: target/linux failed to build.
make[2]: *** [target/Makefile:30: target/linux/compile] Error 1
make[2]: Leaving directory '/scratch/union'
make[1]: *** [target/Makefile:23: /scratch/union/staging_dir/target-mipsel_24kc_musl/stamp/.target_compile] Error 2
make[1]: Leaving directory '/scratch/union'
make: *** [/scratch/union/include/toplevel.mk:231: world] Error 2
```

Can someone take a look/propose change to fix?  @hauke @robimarko ?

graysky2 committed:
kernel: bump 5.15 to 5.15.103

Manually rebased:
        backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch

Removed upstreamed:
        backport-5.15/060-v6.0-01-tools-build-Add-feature-test-for-init_disassemble_in.patch[1]
        backport-5.15/060-v6.0-02-tools-include-add-dis-asm-compat.h-to-handle-version.patch[2]
        backport-5.15/060-v6.0-03-tools-perf-Fix-compilation-error-with-new-binutils.patch[3]
        backport-5.15/060-v6.0-04-tools-bpf_jit_disasm-Fix-compilation-error-with-new-.patch[4]
        backport-5.15/060-v6.0-05-tools-bpftool-Fix-compilation-error-with-new-binutil.patch[5]
        pending-5.15/733-02-net-ethernet-mtk_eth_soc-fix-RX-data-corruption-issu.patch[6]
        bcm47xx/patches-5.15/170-bgmac-fix-initial-chip-reset-to-support-BCM5358.patch[7]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.103&id=51b99dc38c1a053e2e732d7f9e2740e343ae7eae
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.103&id=451c9d7b16169645ed291ebb2ca9844caa088f2d
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.103&id=97f005c0bdbaf656a7808586d234965385a06c58
4. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.103&id=1c27fab243333821375e4d63128d60093fdbe149
5. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.103&id=4441a90091931fd81607567961dc122f24f735bb
6. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.103&id=2adc29350a5b4669544566f71f208d2abaec60ab
7. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.103&id=04bfc5bcdfc0fdb73587487c71b04d63807ae15a

Build system: x86_64
Build-tested: bcm2711/RPi4B, ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: bcm2711/RPi4B, ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod

Signed-off-by: John Audia <therealgraysky@proton.me>

hauke commented on target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch:
```diff
@@ -25,15 +25,16 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
 -	mcr_new |= MAC_MCR_IPG_CFG | MAC_MCR_FORCE_MODE |
--		   MAC_MCR_BACKOFF_EN | MAC_MCR_BACKPR_EN | MAC_MCR_FORCE_LINK;
+-		   MAC_MCR_BACKOFF_EN | MAC_MCR_BACKPR_EN | MAC_MCR_FORCE_LINK |
+-		   MAC_MCR_RX_FIFO_CLR_DIS;
```
You have to add `MAC_MCR_RX_FIFO_CLR_DIS` to the block where it was moved to about 20 lines below this line.

graysky2 replied:
I not sure exactly what you're asking me to add... would you mind sending a mergable snippet here as you did last time?

hauke replied:
This is missing:
````
--- a/target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch
+++ b/target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch
@@ -43,7 +43,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  	return;
  
  err_phy:
-@@ -478,6 +467,26 @@ init_err:
+@@ -478,6 +467,27 @@ init_err:
  		mac->id, phy_modes(state->interface), err);
  }
  
@@ -58,7 +58,8 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
 +	mcr_cur = mtk_r32(mac->hw, MTK_MAC_MCR(mac->id));
 +	mcr_new = mcr_cur;
 +	mcr_new |= MAC_MCR_IPG_CFG | MAC_MCR_FORCE_MODE |
-+		   MAC_MCR_BACKOFF_EN | MAC_MCR_BACKPR_EN | MAC_MCR_FORCE_LINK;
++		   MAC_MCR_BACKOFF_EN | MAC_MCR_BACKPR_EN | MAC_MCR_FORCE_LINK |
++		   MAC_MCR_RX_FIFO_CLR_DIS;
 +
 +	/* Only update control register when needed! */
 +	if (mcr_new != mcr_cur)
````

graysky2 replied:
Thank you for the patch, applied.

graysky2 commented:
I repeated the build-test/run-test for all three in my original commit message.  All are good.

graysky2 commented:
Some recent commits must have caused this breakage (27 failing and 198 successful).  I am re-running the update and will force push.  Should take care of it.

graysky2 committed:
kernel: bump 5.15 to 5.15.104

All patches automatically rebased.

Build system: x86_64
Build-tested: bcm2711/RPi4B, ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: bcm2711/RPi4B, ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod

Signed-off-by: John Audia <therealgraysky@proton.me>
